### PR TITLE
Add scripting around the development environment

### PR DESCRIPTION
**Pull request by markgoddard**
The following top level scripts are added under dev/:

* overcloud-deploy.sh
* seed-deploy.sh
* seed-hypervisor-deploy.sh

Some amount of configuration is possible via dev/config.sh.

### CODE REVIEW

File: doc/source/development/automated.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+-----
+
+SSH into the Vagrant VM::
+
+    vagrant ssh
+
+Run the ``dev/overcloud-deploy.sh`` script to deploy the OpenStack control
+plane::
+
+    /vagrant/kayobe/dev/overcloud-deploy.sh
```
Comment (yankcrime) [inline doc/source/development/automated.rst]: This path should be `/vagrant/dev/overcloud-deploy.sh`

Comment (markgoddard) [reply doc/source/development/automated.rst]: Good spot.

File: dev/environment-setup.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -e
```
Comment (dougszumski) [inline dev/environment-setup.sh]: Add ```set -u``` ? Same comment for other scripts.

File: dev/functions [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    else
+        echo "Pulling overcloud container images"
+        kayobe overcloud container image pull
+    fi
+
+    echo "Deploying containerised overcloud services"
+    kayobe overcloud service deploy
+
+    echo "Performing post-deployment configuration"
+    source ${KOLLA_CONFIG_PATH:-/etc/kolla}/admin-openrc.sh
```
Comment (dougszumski) [inline dev/functions]: nit: add double quotes in-case someone puts a space in their config path? Same for other vars.

File: doc/source/development/vagrant.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+* VirtualBox
+* VMWare Fusion
+
+The VM is configured with 4GB RAM. It has a single private network in addition
+to the standard Vagrant NAT network.
+
+Preparation
+===========
+
+First, ensure that Vagrant is installed and correctly configured to use
```
Comment (dougszumski) [inline doc/source/development/vagrant.rst:21]: Presumably we could also use VMWare Fusion as noted above? Should we mention that a Vagrant provider should be installed and configured at this stage as well?

File: doc/source/development/vagrant.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+virtual box. Also install the following vagrant plugins::
+
+    vagrant plugin install vagrant-vbguest
+    vagrant plugin install vagrant-reload
+
+Note: if using Ubuntu 16.04 LTS, you may be unable to install any plugins. To
```
Comment (dougszumski) [inline doc/source/development/vagrant.rst]: FWIW, i found another workaround which didn't require using the upstream version:
```sudo sed -i'' "s/Specification.all = nil/Specification.reset/" /usr/lib/ruby/vendor_ruby/vagrant/bundler.rb```

**Review by dougszumski — APPROVED** [review]
Looks good to me. I started setting it up, but I hit an issue with having to sign the vbox kernel module, so I've parked it for now.

File: dev/environment-setup.sh (hunk 2 above)
Comment (dougszumski) [reply dev/environment-setup.sh]: I suppose we could also add bashate?

Comment (markgoddard) [reply dev/environment-setup.sh]: Done and done.

Comment (markgoddard) [reply dev/environment-setup.sh]: Also pipefail.

File: dev/functions (hunk 3 above)
Comment (markgoddard) [reply dev/functions]: Now more quotey.

File: doc/source/development/vagrant.rst (hunk 4 above)
Comment (markgoddard) [reply doc/source/development/vagrant.rst:21]: Done.

**Pull request merged**